packages/mui-material/src/styles/unstable_createTheme.d.ts

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { ThemeOptions as SystemThemeOptions, Theme as SystemTheme } from '@mui/system';
+import { OverridableStringUnion } from '@mui/types';
+import { Mixins, MixinsOptions } from './createMixins';
+import { Palette, PaletteOptions } from './createPalette';
+import { Typography, TypographyOptions } from './createTypography';
+import { Shadows } from './shadows';
+import { Transitions, TransitionsOptions } from './createTransitions';
+import { ZIndex, ZIndexOptions } from './zIndex';
+import { Components } from './components';
+
+/**
+ * default MD color-schemes
+ */
+export type DefaultColorScheme = 'light' | 'dark';
+
+/**
+ * The application can add more color-scheme by extending this interface via module augmentation
+ *
+ * Ex.
+ * declare module @mui/material/styles {
+ *   interface ColorSchemeOverrides {
+ *     foo: true;
+ *   }
+ * }
+ *
+ * // SupportedColorScheme = 'light' | 'dark' | 'foo';
+ */
+export interface ColorSchemeOverrides {}
+export type ExtendedColorScheme = OverridableStringUnion<never, ColorSchemeOverrides>;
+
+/**
+ * All color-schemes that the application has
+ */
+export type SupportedColorScheme = DefaultColorScheme | ExtendedColorScheme;
+
+export interface ThemeOptions extends Omit<SystemThemeOptions, 'zIndex'> {
+  mixins?: MixinsOptions;
+  components?: Components<BaseTheme>;
+  // palette?: PaletteOptions;
+  colorSchemes?: Record<SupportedColorScheme, { palette: PaletteOptions }>;
+  shadows?: Shadows;
+  transitions?: TransitionsOptions;
+  typography?: TypographyOptions | ((palette: Palette) => TypographyOptions);
+  zIndex?: ZIndexOptions;
+  unstable_strictMode?: boolean;
+  opacity?: {
+    active?: number;
+    hover?: number;
+    selected?: number;
+    disabled?: number;
+    focus?: number;
+  };
+}
+
+interface BaseTheme extends SystemTheme {
+  mixins: Mixins;
+  palette: Palette;
+  shadows: Shadows;
+  transitions: Transitions;
+  typography: Typography;
+  zIndex: ZIndex;
+  unstable_strictMode?: boolean;
+  colorSchemes: Record<string, { palette: Palette }>;
```

Comment:
```suggestion
```
The final theme does not have `colorSchemes`.

Reply:
My bad, the `colorSchemes` is added to the theme. Now, I remember the use case where I want to list out all the `colorSchemes` from the theme.